spyder/widgets/sourcecode/base.py

```diff
@@ -793,25 +793,19 @@ def go_to_previous_cell(self):
         cursor = self.textCursor()
         cur_pos = prev_pos = cursor.position()
 
-        # Move to the begining of the cell
-        while not self.is_cell_separator(cursor):
-            # Moving to the previous code cell
+        if self.is_cell_separator(cursor):
+            # Move to the previous cell

```

Comment:
Maybe explain the behavior in the comment, not always will move to the previous cell: "Move to the previous cell or the beginning of the current cell"